addons/mail/static/src/models/message_sender/message_sender.js

```diff
@@ -0,0 +1,278 @@
+
+        /**
+         * @private
+         * @param {string} htmlString
+         * @returns {string}
+         */
+        _generateEmojisOnHtml(htmlString) {
+            for (const emoji of emojis) {
+                for (const source of emoji.sources) {
+                    const escapedSource = String(source).replace(
+                        /([.*+?=^!:${}()|[\]/\\])/g,
+                        '\\$1');
+                    const regexp = new RegExp(
+                        '(\\s|^)(' + escapedSource + ')(?=\\s|$)',
+                        'g');
+                    htmlString = htmlString.replace(regexp, '$1' + emoji.unicode);
+                }
+            }
+            return htmlString;
+        }
+
+        async _postMailChannel(postData) {
+            let messageId;
+            const command = this.thread.composer._getCommandFromText(postData.body);
+            Object.assign(postData, {
+                subtype_xmlid: 'mail.mt_comment',
+            });
+            if (command) {
+                messageId = await this.async(() => this.env.models['mail.thread'].performRpcExecuteCommand({
+                    channelId: this.thread.id,
+                    command: command.name,
+                    postData,
+                }));
+            } else {
+                messageId = await this.async(() =>
+                    this.env.models['mail.thread'].performRpcMessagePost({
+                        postData,
+                        threadId: this.thread.id,
+                        threadModel: this.thread.model,
+                    })
+                );
+            }
+            return messageId;
+        }
+
+        async _postMailThread(postData) {
+            Object.assign(postData, {
+                subtype_xmlid: this.thread.composer.isLog ? 'mail.mt_note' : 'mail.mt_comment',
+            });
+            if (!this.thread.composer.isLog) {
+                postData.context = {
+                    mail_post_autofollow: true,
+                };
+            }
+            const messageId = await this.async(() =>
+                this.env.models['mail.thread'].performRpcMessagePost({
+                    postData,
+                    threadId: this.thread.id,
+                    threadModel: this.thread.model,
+                })
+            );
+            const [messageData] = await this.async(() => this.env.services.rpc({
+                model: 'mail.message',
+                method: 'message_format',
+                args: [[messageId]],
+            }, { shadow: true }));
+            this.env.models['mail.message'].insert(Object.assign(
+                {},
+                this.env.models['mail.message'].convertData(messageData),
+                {
+                    originThread: [['insert', {
+                        id: this.thread.id,
+                        model: this.thread.model,
+                    }]],
+                })
+            );
+            this.thread.loadNewMessages();
```

Comment:
Isn't this part also relevant to channels?

Reply:
Doesn't look that way in the master branch : 
https://github.com/odoo/odoo/blob/master/addons/mail/static/src/models/composer/composer.js#L266

Reply:
Well actually the message should be coming back on the bus with proper info anyway so this extra rpc + insert looks like useless duplicate code.

Reply:
Oh yeah that's why it is there... It's because it does come on the bus for channel, but not for other documents.

Reply:
I guess we need this code then ? Or we could fix the bus ?

Reply:
Fine to keep this code in the PR (maybe with an extra comment explaining why). "Fixing" bus for threads is part of another task, that we might or might never do, depending in part on websocket for performance.